server/index.js

```diff
@@ -60,6 +60,90 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart/', (req, res, next) => {
+  const { cartId } = req.session;
+  const sql = `
+      select "c"."cartItemId",
+      "c"."price",
+      "p"."productId",
+      "p"."image",
+      "p"."name",
+      "p"."shortDescription"
+  from "cartItems" as "c"
+  join "products" as "p" using("productId")
+  where "c"."cartId" = $1
+  `;
+  const params = [cartId];
+  db.query(sql, params)
+    .then(result => {
+      if (!req.session.cartId) { res.status(200).json([]); }
```

Comment:
You should do this before trying to query the database with a potentially non-existent `cartId`.